cEP-0017.md

```diff
@@ -0,0 +1,391 @@
+        """
+        if self.version:
+            return "{}{} on {}".format(self.package,
+                                       self.version,
+                                       self.platform)
+        else:
+            return "{} on {}".format(self.package,
+                                     self.platform)
+
+    def install_package(self):
+        """
+        Runs the install command for the package given in a sub-process.
+        """
+        run(" ".join(self.install_command()), stdout=Capture(),
+            stderr=Capture())
+
+    def install_command(self):
+        """
+        Returns a string with the installation command, to be used by
+        "install_package()".
+
+        :raises NotImplementedError: Method is not implemented
+        """
+        raise NotImplementedError
+
+    def is_installed(self):
+        """
+        Check if the requirement is satisfied.
+
+        :return: Returns True if satisfied, False if not.
+        :raises NotImplementedError: Method is not implemented
+        """
+        raise NotImplementedError
+
+    @classmethod
+    @lru_cache()
+    def get_platform():
+        return platform.system(), distro.id()
+
+    def check_platform():
+        """
+        Check if the user is on the target platform
+
+        :return: Returns True if user is on correct platform for this Requirement,
+        otherwise false.
+        """
+        if get_platform()[0] is 'Darwin':
+            return self.platform == 'Darwin'
+        elif get_platform()[0] is 'Linux':
+            return get_platform()[1] == self.platform
+```
+
+## Ansible
+
+### Package Requirement
+
+```yaml
+- pip:
+    name: package_name
+    version: 0.11
+```
+
+This can be used to install the `pip` dependencies for Bears. Ansible currently
+supports the following [language specific packaging modules](http://docs.ansible.com/ansible/latest/modules/list_of_packaging_modules.html#language).
+
+### Distribution Requirement
+
+#### Package name is same across different OS families
+
+```yaml
+- name: Install foo
+  package: name=foo state=latest
+```
+
+Ansible provides a generic and abstracted OS `package` module that can be
+used to install the required package on all distributions. For example for the
+`ArtisticStyleBear` that uses a `DistributionRequirement` as shown:
+
+```python
+REQUIREMENTS = {
+        DistributionRequirement(
+            apt_get='astyle',
+            dnf='astyle'
+        )
+    }
+```
+
+Then indent can be installed simply as follows
+
+```yaml
+- name: Install astyle
+  package: name=astyle state=latest
+```
+
+#### When the package name is different across OS families
+
+```yaml
+# Load a variable file based on the OS type, or use default if not found.
+with_first_found:
+    - "../vars/{{ ansible_distribution }}.yml"
+    - "../vars/{{ ansible_os_family }}.yml"
+    - "../vars/default.yml"
+
+- name: Install bear_name
+  package: >
+    name={{ bear_name }}
```

Comment:
`bear_name` isnt a good variable name for a dependency.

`{{<program_name>}}` would be more intuitive.  
e.g.

```
name={{yamllint}}
```

Also, bears can have multiple dependencies, so any 1:1 relationship in names should be avaoided.


and if we ever add the second yamllint (ruby iirc) in a second bear, then we can simply use a different name, without it looking too weird.

```
name={{ruby_yamllint}}
```

Reply:
yes that is a mistake
it should have been {{ package_name }}